src/app.js

```diff
@@ -59,7 +59,11 @@ function listen (koaApp, config, ledgers) {
   // Start a coroutine that connects to the backend and
   // subscribes to all the ledgers in the background
   co(function * () {
-    yield backend.connect()
+    const success = yield backend.connect()
+    if (!success) {
+      log('app').error('backend initialization has failed - the connector will now exit')
+      process.exit(1)
```

Comment:
Because I wanted all the calls to complete so that we can show an error message for *all* missing currency pairs (as apposed to only the first one -- that would happen if we throw an exception as soon as one call fails.